R/render-plot.R

```diff
@@ -235,6 +235,22 @@ drawPlot <- function(name, session, func, width, height, alt, pixelratio, res, .
   #  9. Return img, value, displaylist, coordmap
   # 10. On error, take width and height dependency
 
+  checkDimsForNulls <- function(width, height) {
+    null_width <- is.null(width)
+    null_height <- is.null(height)
+    if (null_width || null_height) {
+      stop("null ",
+           paste0(c("width", "height")[c(null_width, null_height)], collapse=" and "),
+           " seen for plot output ",
+           name,
+           ". Make sure that the plot container has non-zero size.")
+    }
```

Comment:
I wonder if it would be more sensible to `req()` these values?

Reply:
I think `req` would miss out debug info that is actually quite useful. 

I don't think Shiny authors should ever actually see this error in "normal development" - in normal development there should never by NULL height and widths - I think the Shiny client should always be providing them for visible outputs? 

... although there might be some problem if suspendWhenHidden has been set to `FALSE`? Not sure what should happen in that case - should the JS be sending sizes for those plot outputs?